src/drawers/drawer.ts

```diff
@@ -76,7 +76,14 @@ export class Drawer {
    * @param{any[]} data The data to be drawn
    */
   private _bindSelectionData(data: any[]) {
-    let dataElements = this.selection().data(data);
+    // if the dataset has a key, use it when binding the data   
+    let dataElements: d3.selection.Update<any>;
+    if (this._dataset && this._dataset.keyFunction()) {
+      dataElements = this.selection().data(data, this._dataset.keyFunction());
+    } else {
+      dataElements = this.selection().data(data);
+    }
```

Comment:
I gets errors in the test suite without the test  for Dataset
Abstract Drawer drawer timing works as expected for null animators:
Abstract Drawer drawer timing works for non-null animators:

I suspect that the test just sets up an unrealistic situation, but it does demonstrate that its possible to pass the Dataset as null to the constructor of Drawer.